modules/styleguide/assets/js/frontend/contexts/settings.js

```diff
@@ -0,0 +1,164 @@
+import { createContext, useContext, useEffect, useState, useRef } from 'react';
+import useDebouncedCallback from '../hooks/use-debounced-callback';
+
+const SettingsContext = createContext( null );
+
+/**
+ * @return {{settings: Map, isReady: boolean}|null} context
+ */
+export const useSettings = () => {
+	return useContext( SettingsContext );
+};
+
+export const SettingsProvider = ( props ) => {
+	const [ status, setStatus ] = useState( 'idle' );
+	const [ settings, _setSettings ] = useState( new Map() );
+	const settingsRef = useRef( settings );
+
+	const setSettings = ( newSettings ) => {
+		settingsRef.current = newSettings;
+		_setSettings( newSettings );
+	};
```

Comment:
Why is it `ref`?

Reply:
To be able to work with the state through the event listener. It's a workaround for that.

Reply:
Can this be resolved?